pallets/payroll/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+        #[pallet::weight((<T as Config>::WeightInfo::create_stream(), DispatchClass::Operational))]
+        #[transactional]
+        pub fn create_stream(
+            origin: OriginFor<T>,
+            recipient: AccountOf<T>,
+            deposit: BalanceOf<T>,
+            currency_id: AssetIdOf<T>,
+            rate_per_sec: BalanceOf<T>,
+            start_time: Timestamp,
+            stop_time: Timestamp,
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+            ensure!(recipient != sender, Error::<T>::StreamToOrigin);
+            ensure!(deposit > 0, Error::<T>::DepositIsZero);
+            ensure!(
+                start_time >= T::UnixTime::now().as_secs(),
+                Error::<T>::StartBeforeBlockTime
+            );
+            ensure!(stop_time > start_time, Error::<T>::StopBeforeStart);
+            // insert stream to the Streams
+            let stream: Stream<T> = Stream {
+                remaining_balance: deposit, // remaining balance same value for now due to initialization
+                deposit,                    // deposit
+                currency_id,                // currency id
+                rate_per_sec,               // rate per second
+                recipient: recipient.clone(),                  // recipient
+                sender: sender.clone(),                     // sender
+                start_time,                 // start_time
+                stop_time,                  // stop_time
+            };
+            let stream_id = NextStreamId::<T>::get();
+            // Insert stream to runtime
+            Streams::<T>::insert(stream_id.clone(), stream);
+            // Increment stream id
+            NextStreamId::<T>::set(stream_id + 1);
+            // transfer deposit from sender to global EOA
+            T::Assets::transfer(currency_id.clone(), &sender, &Self::account_id(), deposit.clone(), false)?;
+            Self::deposit_event(Event::<T>::CreateStream(stream_id, sender, recipient, deposit, currency_id, start_time, stop_time));
+            Ok(().into())
+        }
+
+        /// Cancel an existing stream
+        #[pallet::weight((<T as Config>::WeightInfo::cancel_stream(), DispatchClass::Operational))]
+        #[transactional]
+        pub fn cancel_stream(
+            origin: OriginFor<T>,
+            stream_id: StreamId,
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+            // check sender is stream sender
+            let stream = Streams::<T>::get(stream_id.clone()).ok_or(DispatchError::CannotLookup)?;
+            ensure!(
+                sender == stream.sender,
+                Error::<T>::NotTheStreamer
+            );
+            // get sender and recipient balance at result
+            let sender_balance =  Self::balance_of(stream.clone(), &sender)?;
+            let recipient_balance = Self::balance_of(stream.clone(), &stream.recipient)?;
```

Comment:
Can we remove `clone()` here?

Reply:
I used borrows in internal functions that does not change values to lower memory usage.